src/machine.tex

```diff
@@ -1830,6 +1830,54 @@ \subsection{Machine Bad Bits ({\tt mtval}) Register}
 
 \section{Machine-Mode Privileged Instructions}
 
+\subsection{Environment Call and Breakpoint}
+
+\vspace{-0.2in}
+\begin{center}
+\begin{tabular}{M@{}R@{}F@{}R@{}S}
+\\
+\instbitrange{31}{20} &
+\instbitrange{19}{15} &
+\instbitrange{14}{12} &
+\instbitrange{11}{7} &
+\instbitrange{6}{0} \\
+\hline
+\multicolumn{1}{|c|}{funct12} &
+\multicolumn{1}{c|}{rs1} &
+\multicolumn{1}{c|}{funct3} &
+\multicolumn{1}{c|}{rd} &
+\multicolumn{1}{c|}{opcode} \\
+\hline
+12 & 5 & 3 & 5 & 7 \\
+ECALL   & 0 & PRIV & 0 & SYSTEM \\
+EBREAK  & 0 & PRIV & 0 & SYSTEM \\
+\end{tabular}
+\end{center}
+
+The ECALL instruction is used to make a request to the supporting execution
+environment.  When executed in U-mode, S-mode, or M-mode, it generates an
```

Comment:
"Generates an environment call" is a little bit vague.  Would something more like «generates an exception of type "environment call from U-mode", "environment call from S-mode", or "environment call from M-mode", respectively» work?

Also, would it make sense to add a commentary to the user spec saying that the runtime behavior is documented elsewhere?